### PR TITLE
Expo support

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

## Changes

* Define expo plugin
* Remove `adyenReactNativeRedirectScheme `
* Update README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,26 +135,17 @@ protected void onCreate(Bundle savedInstanceState) {
...
-```
-
-2. Add `intent-filter` to your Checkout activity:
+1. Add `intent-filter` to your Checkout activity:
 ```xml
 <intent-filter>
     <action android:name="android.intent.action.VIEW" />
     <category android:name="android.intent.category.DEFAULT" />
     <category android:name="android.intent.category.BROWSABLE" />
-    <data android:host="${applicationId}" android:scheme="${redirectScheme}" />
```
Comment (Robert-SD) [inline README.md:92]: Is defining the host not necessary anymore?

File: plugin/src/withAdyenIos.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    appDelegateModResults.contents = setImport(
+      appDelegateModResults.contents,
+      useFrameworks
+    );
+    appDelegateModResults.contents = setRedirectComponent(
+      appDelegateModResults.contents
+    );
+    return newConfig;
+  });
+  // apply Apple Pay Merchant ID
```
Comment (Robert-SD) [inline plugin/src/withAdyenIos.ts]: Shall this comment stay for better explanation?

**Review by Robert-SD — DISMISSED** [review]

File: docs/v2-MigrationGuide.md [hunk 3]
```diff
@@ -2,10 +2,13 @@
...
 
+### Android
+
 * Merchant's app theme must be decendent of `Theme.MaterialComponents` to operate with "instant" payment components (ex. Paypal, Klarna). Example:
 ```xml
     <style name="AppTheme" parent="Theme.MaterialComponents.DayNight.NoActionBar"> 
 ```
+* `adyenReactNativeRedirectScheme` was deprecared. Use any [intentFilter](https://developer.android.com/guide/components/intents-filters). SDK will still provides `returnUrl` value inside of the `onSubmit.data` in case it is needed, be cautious to not override it.
 
 ## Non breacking changes
```
Comment (atmamont) [inline docs/v2-MigrationGuide.md]: breacking -> breaking

File: plugin/src/withAdyen.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import {
+  ConfigPlugin,
+  createRunOncePlugin
+} from '@expo/config-plugins';
+import { withAdyenAndroid } from './withAdyenAndroid';
+import { withAdyenIos } from './withAdyenIos';
```
Comment (atmamont) [inline plugin/src/withAdyen.ts:6]: It feels something wrong with iOS being spelled as `Ios` :)

**Review by atmamont — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (descorp) [reply README.md:92]: It is up to merchant. Host is needed to filter intents out (e.x. for DropIn Activity), and we no longer filtering anything.

File: docs/v2-MigrationGuide.md (hunk 3 above)
Comment (descorp) [reply docs/v2-MigrationGuide.md]: Fixed [dde68d8](https://github.com/Adyen/adyen-react-native/pull/397/commits/dde68d87d2f1ad56337188a829c80a1e06850af9)

**Review by Robert-SD — DISMISSED** [review]

File: plugin/src/withAdyen.ts (hunk 4 above)
Comment (descorp) [reply plugin/src/withAdyen.ts:6]: I know 🙈
But his is a naming convention, apparently.. 

Used by [Expo](https://github.com/expo/expo/blob/e882f89effb0224ed45fd846a6d7fbc0a7eaee06/packages/%40expo/cli/src/utils/scheme.ts#L35), [FireBase](https://github.com/invertase/react-native-firebase/blob/bb241c6b235bd50e4e2aaefce94d9d8b66233a9a/packages/app/plugin/src/index.ts#L8), [Other plugins](https://github.com/react-native-voice/voice/blob/32aab8b00e57132fa3de204f86d75635c969e5d4/plugin/src/withVoice.ts#L34) ...

Comment (atmamont) [reply plugin/src/withAdyen.ts:6]: Ouch!

**Review by atmamont — DISMISSED** [review]

File: plugin/src/withAdyenIos.ts (hunk 2 above)
Comment (descorp) [reply plugin/src/withAdyenIos.ts]: Updated

**Review by Robert-SD — APPROVED** [review]

**Pull request merged**